src/util/Vocabularies.ts

```diff
@@ -2,57 +2,81 @@
 import { DataFactory } from 'n3';
 import type { NamedNode } from 'rdf-js';
 
-type RecordOf<TKey extends any[], TValue> = Record<TKey[number], TValue>;
+/**
+ * A `Record` in which each value is a concatenation of the baseUrl and its key.
+ */
+type ExpandedRecord<TBase extends string, TLocal extends string> = {[K in TLocal]: `${TBase}${K}` };
 
-export type Namespace<TKey extends any[], TValue> =
-  { namespace: TValue } & RecordOf<TKey, TValue>;
+/**
+ * Has a base URL as `namespace` value and each key has as value the concatenation with that base URL.
+ */
+export type Namespace<TBase extends string, TLocal extends string> =
+  { namespace: TBase } & ExpandedRecord<TBase, TLocal>;
```

Comment:
I find it a bit quirky that a Namespace has a namespace. Perhaps
```suggestion
  { base: TBase } & ExpandedRecord<TBase, TLocal>;
```